packages/components/inputs/select-utils/src/clear-indicator/clear-indicator.tsx

```diff
@@ -5,20 +5,22 @@ import { useIntl } from 'react-intl';
 import { designTokens } from '@commercetools-uikit/design-system';
 import { CloseIcon } from '@commercetools-uikit/icons';
 import messages from './messages';
+import { TSelectInputCustomComponentProps } from '../types';
 
 type TInnerProps = {
   ref: LegacyRef<HTMLButtonElement>;
 } & JSX.IntrinsicElements['button'];
 
 export type TClearIndicatorProps = {
   innerProps: TInnerProps;
-} & ClearIndicatorProps;
+} & TSelectInputCustomComponentProps<ClearIndicatorProps>;
```

Comment:
Maybe we could simply use the same pattern as seen here: 
https://github.com/commercetools/ui-kit/blob/c2447dbae4048b2a06017c6ef3db88efdbc955fd/packages/components/inputs/select-utils/src/tag-remove/tag-remove.tsx#L9-L15 for typing the `selectProps` prop

Reply:
I think it would be a better approach what we are proposing here as not only known custom components but the ones provided by consumers would need to extend the typing.

Please, reach out to me and we can discuss this in a meeting if you think we're going in the wrong direction.